playbooks/provisioning/openstack/prerequisites.yml

```diff
@@ -0,0 +1,72 @@
+---
+- hosts: localhost
+  tasks:
+  # Check ansible
+  - name: Get Ansible version
+    shell: >
+      ansible --version | head -n 1 |
+      while read a b; do echo $b; done | head -c 3
+    register: ansible_result
+  - name: Check Ansible version
+    assert:
+      that: '{{ ansible_result.stdout }} >= 2.3'
+      msg: "Ansible version must be at least 2.3"
+
+  # Check shade
+  - name: Try to import python module shade
+    command: python -c "import shade"
+    ignore_errors: yes
+    register: shade_result
+  - name: Check if shade is installed
+    assert:
+      that: '{{ shade_result.rc }} == 0'
+      msg: "Python module shade is not installed"
+
+  # Check python-dns
+  - name: Try to import python DNS module
+    command: python -c "import dns"
+    register: pythondns_result
+  - name: Check if python-dns is installed
+    assert:
+      that: '{{ pythondns_result.rc }} == 0'
+      msg: "Python module python-dns is not installed"
+
+  # Check jinja2
+  - name: Try to import jinja2 module
+    command: python -c "import jinja2"
+    ignore_errors: yes
+    register: jinja_result
+  - name: Check if jinja2 is installed
+    assert:
+      that: '{{ jinja_result.rc }} == 0'
+      msg: "Python module jinja2 is not installed"
+
+  # Check Glance image
+  - name: Try to show image
+    shell: openstack image show {{ openstack_default_image_name }}
```

Comment:
Please take a look at this module: http://docs.ansible.com/ansible/os_image_facts_module.html

If it doesn't work, use the `command` module here instead of `shell`. We don't do any piping here so it's not necessary.